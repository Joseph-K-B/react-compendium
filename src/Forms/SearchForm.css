.searchForm{
    display: flex;
    flex-direction: column;
    text-align: center;
    align-items: center;
    justify-content: center;
    border: solid blue 0.25rem;
}

.inputDiv{
    border: solid white 0.15rem;
}

button{
    margin: 10.5%;
}



