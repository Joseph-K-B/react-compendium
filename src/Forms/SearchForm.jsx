
function SearchForm({ name, handleSubmit, handleSearchChange }) {
    return (
        <form onSubmit={handleSubmit} aria-label='searchForm'>
            <h3>Search</h3>
            <label htmlFor='search'>Name:</label>
            <input 
            type='text'
            aria-label='inputField'
            onChange={(e) => handleSearchChange(e.target.value)}
            value={name}
            />
            <button aria-label='button'>Submit</button>
        </form>
    )
}

export default SearchForm