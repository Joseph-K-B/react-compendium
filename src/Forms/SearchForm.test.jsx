import { render, screen } from '@testing-library/react';
import SearchForm from './SearchForm';

it('should render input field and button for search', () => {
    const {container} = render(<SearchForm />);
    const searchForm = screen.getByLabelText(/searchForm/);
    const inputField = screen.getByLabelText(/inputField/);
    const button = screen.getByLabelText(/button/);

    expect(searchForm).toBeInTheDocument();
    expect(inputField).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})