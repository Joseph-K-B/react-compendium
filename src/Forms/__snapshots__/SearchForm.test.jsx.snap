// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render input field and button for search 1`] = `
<div>
  <form
    aria-label="searchForm"
  >
    <h3>
      Search
    </h3>
    <label
      for="search"
    >
      Name:
    </label>
    <input
      aria-label="inputField"
      type="text"
      value=""
    />
    <button
      aria-label="button"
    >
      Submit
    </button>
  </form>
</div>
`;
